From 19c53bd8878e3fa680c8d89a83faacb73ce4cdde Mon Sep 17 00:00:00 2001
From: David Mulder via samba-technical <samba-technical@lists.samba.org>
Date: Wed, 7 Jun 2017 06:43:07 -0600
Subject: [PATCH] messaging: fix net command failure due to unhandled return
 code

messaging_init_internal() blanket returned NT_STATUS_INTERNAL_ERROR
instead of correctly changing the return code to an NTSTATUS code. Also
return more appropriate mem error.

Bug: https://bugzilla.samba.org/show_bug.cgi?id=12828

Signed-off-by: David Mulder <dmulder@suse.com>
Reviewed-by: Jeremy Allison <jra@samba.org>
Reviewed-by: David Disseldorp <ddiss@samba.org>

Autobuild-User(master): David Disseldorp <ddiss@samba.org>
Autobuild-Date(master): Thu Jun  8 08:04:05 CEST 2017 on sn-devel-144

(cherry picked from commit 08a21f3539fef76a22189b1751fd2a081937a057)
---
 source3/lib/messages.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/source3/lib/messages.c b/source3/lib/messages.c
index 533e8694612..69dfbf30b67 100644
--- a/source3/lib/messages.c
+++ b/source3/lib/messages.c
@@ -302,7 +302,7 @@ static NTSTATUS messaging_init_internal(TALLOC_CTX *mem_ctx,
 					     &ret);
 	if (ctx->msg_dgm_ref == NULL) {
 		DEBUG(2, ("messaging_dgm_ref failed: %s\n", strerror(ret)));
-		status = NT_STATUS_INTERNAL_ERROR;
+		status = map_nt_error_from_unix(ret);
 		goto done;
 	}
 	talloc_set_destructor(ctx, messaging_context_destructor);
@@ -313,7 +313,7 @@ static NTSTATUS messaging_init_internal(TALLOC_CTX *mem_ctx,
 		if (ret != 0) {
 			DEBUG(2, ("messaging_ctdbd_init failed: %s\n",
 				  strerror(ret)));
-			status = NT_STATUS_INTERNAL_ERROR;
+			status = map_nt_error_from_unix(ret);
 			goto done;
 		}
 	}
@@ -326,7 +326,7 @@ static NTSTATUS messaging_init_internal(TALLOC_CTX *mem_ctx,
 					  TDB_INCOMPATIBLE_HASH|TDB_CLEAR_IF_FIRST);
 	if (ctx->names_db == NULL) {
 		DBG_DEBUG("server_id_db_init failed\n");
-		status = NT_STATUS_INTERNAL_ERROR;
+		status = NT_STATUS_NO_MEMORY;
 		goto done;
 	}
 
-- 
2.12.3

